src/App.js

```diff
@@ -3,27 +3,33 @@ import './App.css';
 import Colors from './colors'
 
 class ColorApp extends Component {
-  state = {
-    color: [
-      {color: this.getRandomColor(),
-       lock: true
-      },
-      {color: this.getRandomColor(),
-        lock: true
-      },
-      {color: this.getRandomColor(),
-        lock: true
-      },
-      {color: this.getRandomColor(),
-        lock: true
-      },
-      {color: this.getRandomColor(),
-        lock: true
-      },
-    ]
+
+  constructor(){
+    super();
+    let state = {
+      colors: [
+        {color: this.getRandomColor(),
+         lock: false
+        },
+        {color: this.getRandomColor(),
+          lock: false
+        },
+        {color: this.getRandomColor(),
+          lock: false
+        },
+        {color: this.getRandomColor(),
+          lock: false
+        },
+        {color: this.getRandomColor(),
+          lock: false
+        },
+      ]
+    }
+    this.state = state
   }
+  
 
-  this.lockButtonToggle = this.lockButtonToggle.bind(this);
+  
```

Comment:
You do not need to bind the function here you are already doing in when you call it with the toggle prop that you passed into the Colors component. (Arrow functions automatically bind )